embassy-net/src/lib.rs

```diff
@@ -363,6 +366,56 @@ impl<D: Driver + 'static> Stack<D> {
         v4_up || v6_up
     }
 
+    /// Wait for the network stack to obtain a valid IP configuration.
+    /// Returns instantly if [`Stack::is_config_up`] returns `true`.
+    ///
+    /// ## Watch out:
+    /// The Future is polled only when the [`Stack`] is running,
+    /// e.g. call `spawner.spawn(net_task(stack))`.
+    ///
+    /// `await`ing before will never yield!
```

Comment:
There is a chance it will never yield anyway, for example in the case of DHCP. Maybe we should reword this to say as much, and that the end user should handle the timeout?

Reply:
That's right. I guess if DHCP does not assign an IP, the user will have to handle it anyway. I'll think of something and maybe add an example.

Reply:
I've rephrased the docstring. Do you think it is better now?